src/lints/pub_enum_should_be_non_exhaustive.ron

```diff
@@ -0,0 +1,51 @@
+SemverQuery(
+    id: "pub_enum_should_be_non_exhaustive",
+    human_readable_name: "Public API enum added without #[non_exhaustive]",
+    description: "A new public API enum has been added without #[non_exhaustive], which may cause future breaking changes.",
+    required_update: Minor,
+    lint_level: Warn,
```

Comment:
Until we have better lint grouping ability, we probably want this to be an allow-by-default lint. Otherwise many of our users will find it to be too noisy.
```suggestion
    lint_level: Allow,
```